backend/corpora/common/entities/dataset.py

```diff
@@ -79,10 +81,25 @@ def create(
 
         return cls(new_db_object)
 
+    def get_asset(self, asset_uuid) -> typing.Union[Asset, None]:
+        """
+        Retrieve the asset if it exists in the dataset.
+        :param asset_uuid: uuid of the asset to find
+        :return: If the asset is found it is returned, else None is returned.
+        """
+        asset = [asset for asset in self.artifacts if asset.id == asset_uuid]
+        if not asset:
+            return
+        else:
+            return Asset(asset[0])
+
     def delete(self):
         """
         Delete the Dataset and all child objects. Contributors connect to a dataset are deleted if they are not longer
         connected to any datasets.
+
+        TODO: Must also delete s3 objects related to the asset.
```

Comment:
I'd hold off on adding TODOs related to deletion until we have sorted out our deletion user stories.

Reply:
removed